core/src/test/scala/unit/kafka/server/FetchRequestTest.scala

```diff
@@ -227,7 +228,7 @@ class FetchRequestTest extends BaseRequestTest {
       val suffix = s"$tp-$messageIndex"
       new ProducerRecord(tp.topic, tp.partition, s"key $suffix", s"value $suffix")
     }
-    records.map(producer.send).foreach(_.get)
+    records.map(producer.send(_).get)
```

Comment:
Is it intentional that we want to wait for a send to complete before doing the next send? Is it related to how many batches we create?